### PR TITLE
RGRIDT-877 - Possibility to have custom column header tooltips

### PR DESCRIPTION
**Pull request by JoaoFerreira-FrontEnd**
### What was happening
* The Column header tooltips were static, displaying the cell content.

### What was done
* Add a parameter to the column config, where the developer can customize the tooltip content.

### Test Steps
1. Fill the new Column Parameter with some text;
2. Check the tooltip hovering the column header; 
3. Fill the new Column Parameter on another Column with html;
4. Check the tooltip content, it shouldn't process the html (security);
5. Fill the new Column Parameter on another Column with an id;
6. Check the tooltip content, it should clone the content of the target id;

### Checklist
* [x] tested locally
* [x] documented the code
* [x] clean all warnings and errors of eslint
* [x] requires changes in OutSystems
* [x] requires new sample page in OutSystems

### CODE REVIEW

File: code/src/GridAPI/ColumnManager.ts [hunk 1]
```diff
@@ -51,6 +51,10 @@ namespace GridAPI.ColumnManager {
                     jsonEditorConfigs.conditionalFormat
                 );
             }
+
+            if (jsonConfigs.headerTooltip) {
+                //
+            }
```
Comment (tmbp95) [inline code/src/GridAPI/ColumnManager.ts:57]: Why do we need this part?

Comment (JoaoFerreira-FrontEnd) [reply code/src/GridAPI/ColumnManager.ts:57]: thanks, I will clean this part

File: code/src/WijmoProvider/Features/ToolTip.ts [hunk 2]
```diff
@@ -56,25 +56,34 @@ namespace WijmoProvider.Feature {
                     );
                 }
             } else if (ht.cellType === wijmo.grid.CellType.ColumnHeader) {
-                if (
+                const rendered = this._grid.getColumn(ht.getColumn().binding)
+                    .config;
+                this._grid.provider.columns[ht.col] || {};
+
+                if (_currTarget && rendered.headerTooltip) {
+                    if (document.getElementById(rendered.headerTooltip)) {
```
Comment (bmartinho) [inline code/src/WijmoProvider/Features/ToolTip.ts:64]: Can you please add a comment here?

Comment (JoaoFerreira-FrontEnd) [reply code/src/WijmoProvider/Features/ToolTip.ts:64]: setTooltip wijmo method, allows us to render the content of another element (by Id). If an user wants to have a custom content we can "offer" this workaround

Comment (glundgren93) [reply code/src/WijmoProvider/Features/ToolTip.ts:64]: I think Bruno meant to add a comment on the code 😄

Comment (JoaoFerreira-FrontEnd) [reply code/src/WijmoProvider/Features/ToolTip.ts:64]: I did understand ahah

File: code/src/WijmoProvider/Features/ToolTip.ts [hunk 3]
```diff
@@ -56,25 +56,34 @@ namespace WijmoProvider.Feature {
                     );
                 }
             } else if (ht.cellType === wijmo.grid.CellType.ColumnHeader) {
-                if (
+                const rendered = this._grid.getColumn(ht.getColumn().binding)
+                    .config;
+                this._grid.provider.columns[ht.col] || {};
```
[the remark this answers is not in the document]
Comment (rugoncalves) [reply code/src/WijmoProvider/Features/ToolTip.ts:61]: @JoaoFerreira-FrontEnd, let's try to close this conversation. Can you please explain what this line is doing?

[the remark this answers is not in the document]
Comment (JoaoFerreira-FrontEnd) [reply code/src/WijmoProvider/Features/ToolTip.ts:61]: It seems an old solution, I'll delete it.

**Review by glundgren93 — APPROVED** [review]

**Review by rugoncalves — APPROVED** [review]

**Pull request merged**